From 0b170962d33b7244be3144609b5ddc26a5dd041d Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Wed, 28 Jun 2023 15:21:51 +0200
Subject: [PATCH] ARM: dts: stm32: use i2c6 scmi clock on stm32mp15 family

On STM32MP15 soc, I2C4 and I2C6 share the same clock. So if I2C4 clock is
handled by SCMI, I2C6 should be handled by SCMI too.

Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I8e35430d03cae784e3a6260dfd79ad6539500c1d
---
 arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dtsi | 5 +++++
 arch/arm/boot/dts/st/stm32mp157a-ed1-scmi.dtsi | 5 +++++
 arch/arm/boot/dts/st/stm32mp157a-ev1-scmi.dtsi | 5 +++++
 arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dtsi | 5 +++++
 arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dtsi | 5 +++++
 arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dtsi | 5 +++++
 6 files changed, 30 insertions(+)

--- a/arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dtsi
@@ -41,6 +41,11 @@
 	resets = <&scmi_reset RST_SCMI_I2C4>;
 };
 
+&i2c6 {
+	clocks = <&scmi_clk CK_SCMI_I2C6>;
+	resets = <&scmi_reset RST_SCMI_I2C6>;
+};
+
 &iwdg2 {
 	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
 };
--- a/arch/arm/boot/dts/st/stm32mp157a-ed1-scmi.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp157a-ed1-scmi.dtsi
@@ -41,6 +41,11 @@
 	resets = <&scmi_reset RST_SCMI_I2C4>;
 };
 
+&i2c6 {
+	clocks = <&scmi_clk CK_SCMI_I2C6>;
+	resets = <&scmi_reset RST_SCMI_I2C6>;
+};
+
 &iwdg2 {
 	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
 };
--- a/arch/arm/boot/dts/st/stm32mp157a-ev1-scmi.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp157a-ev1-scmi.dtsi
@@ -39,6 +39,11 @@
 	resets = <&scmi_reset RST_SCMI_I2C4>;
 };
 
+&i2c6 {
+	clocks = <&scmi_clk CK_SCMI_I2C6>;
+	resets = <&scmi_reset RST_SCMI_I2C6>;
+};
+
 &iwdg2 {
 	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
 };
--- a/arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dtsi
@@ -46,6 +46,11 @@
 	resets = <&scmi_reset RST_SCMI_I2C4>;
 };
 
+&i2c6 {
+	clocks = <&scmi_clk CK_SCMI_I2C6>;
+	resets = <&scmi_reset RST_SCMI_I2C6>;
+};
+
 &iwdg2 {
 	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
 };
--- a/arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dtsi
@@ -46,6 +46,11 @@
 	resets = <&scmi_reset RST_SCMI_I2C4>;
 };
 
+&i2c6 {
+	clocks = <&scmi_clk CK_SCMI_I2C6>;
+	resets = <&scmi_reset RST_SCMI_I2C6>;
+};
+
 &iwdg2 {
 	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
 };
--- a/arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dtsi
@@ -44,6 +44,11 @@
 	resets = <&scmi_reset RST_SCMI_I2C4>;
 };
 
+&i2c6 {
+	clocks = <&scmi_clk CK_SCMI_I2C6>;
+	resets = <&scmi_reset RST_SCMI_I2C6>;
+};
+
 &iwdg2 {
 	clocks = <&rcc IWDG2>, <&scmi_clk CK_SCMI_LSI>;
 };
